### PR TITLE
Refactored various form field components

### PR DESCRIPTION
**Pull request by jcq**
Connects #14762

### Description
This refactors various form field components so that they support both controlled and uncontrolled variants. It also adds stories and Jest tests for components that didn't already have them.

In addition all now support `inputRef` and `inputProps` props that get passed along to the underlying `input` elements.

### Acceptance Criteria
- [ ] Code compiles correctly
- [ ] `TextField` updated
- [ ] `Textarea` updated
- [ ] `RadioField` updated
- [ ] `Checkbox` updated

### CODE REVIEW

File: client/test/app/components/Checkbox.test.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+  it('renders correctly', async () => {
+    const { container } = setup();
+
+    expect(container).toMatchSnapshot();
+  });
+
+  it('passes a11y testing', async () => {
+    const { container } = setup();
+
+    const results = await axe(container);
```
Comment (sahalliburton) [inline client/test/app/components/Checkbox.test.js:44]: This is really cool, I didnt know you could use axe in the jest context

**Comment by jcq** [comment]
> 1. In storybook for the `HearingConversion` component, I am not able to toggle radio buttons


For the first item, it looks like that's actually the behavior in `master`:
https://www.caseflowdemo.com/storybook/?path=/story/hearings-components-hearingconversion--convert-from-virtual

I think the issue there is that you should be simply passing `hearing` here instead of `props.hearing` — the former gets updated, the latter is static:
https://github.com/department-of-veterans-affairs/caseflow/blob/master/client/app/hearings/components/HearingConversion.stories.js#L43

(You'll also need to put that below where you're spreading `props` or it gets overridden again.)

**Comment by jcq** [comment]
> 2. When shadowing a hearing coordinator and following the workflow below I am unable to see the radio buttons for hearing time which should be displayed
> 
> * Shadow user `BVASYELLOW`
> * Navigate to the hearings schedule and choose a Central hearing: `http://localhost:3000/hearings/schedule`
> * Click the 'Edit hearing Details' link
> * Change the hearing type to a Virtual hearing
> * Checkout the HearingTime component

Similarly here, it seems like things are working the same on `master`, unless I'm missing something? 
https://www.caseflowdemo.com/hearings/66519d5b-316b-4f05-bc20-f0790b80e701/details

From what I see in Storybook (on this branch as well as demo), radio fields should only show up when converting *from*  a virtual hearing rather than *to* one. I tried switching one to virtual in local env and then switching it back and radios did show up (and seemed functional).

**Comment by sahalliburton** [comment]
Thanks for confirming @jcq I will go ahead and create a ticket to address those issues then

**Pull request merged**